addons/mass_mailing/wizard/mail_compose_message.py

```diff
@@ -20,60 +20,83 @@ def get_mail_values(self, res_ids):
         self.ensure_one()
         res = super(MailComposeMessage, self).get_mail_values(res_ids)
         # use only for allowed models in mass mailing
-        if self.composition_mode == 'mass_mail' and \
-                (self.mass_mailing_name or self.mass_mailing_id) and \
-                self.env['ir.model'].sudo().search_count([('model', '=', self.model), ('is_mail_thread', '=', True)]):
+        if self._should_add_traces():
             mass_mailing = self.mass_mailing_id
             if not mass_mailing:
+                active_ids_domain = "['id', 'in', {}]".format(self._context.get('active_ids'))
+                # add active_ids to domain so that it doesn't 're'-send on records that were not selected
+                if self.active_domain and self.active_domain.endswith(']') and self.active_domain.lstrip('[').rstrip(']'):
+                    enhanced_active_domain = self.active_domain[:-1] + ',' + active_ids_domain + ']'
+                else:
+                    enhanced_active_domain = active_ids_domain
```

Comment:
can you use `ast.literal_eval` ? (and then `repr` to convert it back to a string)

Reply:
Note that this may not be needed anymore, I'm pretty sure this was only really needed for the resend wizard logic (which I ended up not including)

I'll add a condition to make sure we only apply this when there is an active_ids key, it seems like it could easily mess with regular mass mailings otherwise. But we can also probably completely remove it if it adds no value.